From 1f98984e75119d5023f117acb89ef4a9de935c73 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Sun, 16 Jan 2022 08:17:41 +0000
Subject: [PATCH 91/94] WIP: arm64: dts: meson: add support for MeCool M8S Pro
 W

MeCool M8S Pro W is an Android STB based on the Amlogic P281 (GXL)
reference design with an S905W chip and the following specs:

- 2GB DDR3 RAM
- 16GB eMMC
- 10/100 Base-T Ethernet
- SSV6051P Wireless (802.11 b/g/n)
- HDMI 2.0 video
- 1x 3.5mm AV jack
- 2x USB 2.0 port
- IR receiver
- 1x Power LED (blue)
- 1x Update/Reset button (underside)
- 1x micro SD card slot

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 .../meson-gxl-s905w-mecool-m8s-pro-w.dts      | 30 +++++++++++++++++++
 1 file changed, 30 insertions(+)
 create mode 100644 arch/arm64/boot/dts/amlogic/meson-gxl-s905w-mecool-m8s-pro-w.dts

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl-s905w-mecool-m8s-pro-w.dts b/arch/arm64/boot/dts/amlogic/meson-gxl-s905w-mecool-m8s-pro-w.dts
new file mode 100644
index 000000000000..a5bbe7315471
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905w-mecool-m8s-pro-w.dts
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2018 Martin Blumenstingl <martin.blumenstingl@googlemail.com>.
+ * Based on meson-gxl-s905d-p231.dts:
+ * - Copyright (c) 2016 Endless Computers, Inc.
+ *   Author: Carlo Caione <carlo@endlessm.com>
+ */
+
+/dts-v1/;
+
+#include "meson-gxl-s905x.dtsi"
+#include "meson-gx-p23x-q20x.dtsi"
+
+/ {
+	compatible = "videostrong,gxl-m8s-pro-w", "amlogic,s905w", "amlogic,meson-gxl";
+	model = "MeCool M8S Pro W";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+};
+
+&ir {
+	linux,rc-map-name = "rc-mecool-m8s-pro-w";
+};
+
+&usb {
+	dr_mode = "host";
+};
-- 
2.35.1

